helpers/utils.js

```diff
@@ -91,11 +94,46 @@ module.exports = {
 
 const playSong = (gameSession, guildPreference, db, message) => {
     let voiceChannel = message.member.voice.channel;
-    const streamOptions = { volume: guildPreference.getVolume(), bitrate: voiceChannel.bitrate };
+    const streamOptions = {
+        volume: guildPreference.getStreamVolume(),
+        bitrate: voiceChannel.bitrate
+    };
+
+    const cacheStreamOptions = {
+        volume: guildPreference.getCachedStreamVolume(),
+        bitrate: voiceChannel.bitrate
+    };
+
+    if (!fs.existsSync(CACHE_DIR)) {
+        fs.mkdirSync(dir)
+    }
+
+    const ytdlOptions = {
+        filter: "audioonly",
+        quality: "highest"
+    };
+
+    const songLocation = `${CACHE_DIR}/${gameSession.getLink()}.mp3`;
+    if (!fs.existsSync(songLocation)) {
+        gameSession.isSongCached = false;
+        const tempLocation = `${songLocation}.part`;
+        if (!fs.existsSync(tempLocation)) {
```

Comment:
What happens if we have a partially completed download that was interrupted?

Reply:
This needs to be fixed with that new issue you told me to create. It'll just stream the song instead, but the `.part` file will exist in an incomplete state.